compiler/src/iree/compiler/InputConversion/Common/QuantizedMatmulToMatmul.cpp

```diff
@@ -88,34 +105,43 @@ struct QuantizedMatmulToMatmul
     int indexOfLhsZpTimesRhsZpTimesKSizeInput = 0;
     Type accElTy = accType.getElementType();
     if (!rhsZpIsConstantZero) {
+      SmallVector<bool> colRedIterator = {false, true};
+      if (batch) {
+        colRedIterator = {false, false, true};
+      };
       Value lhsSums = sumReduceDimensionSubset(builder, lhs, accElTy,
-                                               /*is_reduction=*/{false, true});
+                                               /*is_reduction=*/colRedIterator);
       indexOfLhsSumsInput = addInput(lhsSums, mapToRowDim);
       indexOfRhsZpInput = addInput(rhsZp, mapToNone);
     }
     if (!lhsZpIsConstantZero) {
+      SmallVector<bool> rowRedIterator = {true, false};
+      if (batch) {
+        rowRedIterator = {false, true, false};
+      };
       Value rhsSums = sumReduceDimensionSubset(builder, rhs, accElTy,
-                                               /*is_reduction=*/{true, false});
+                                               /*is_reduction=*/rowRedIterator);
       indexOfRhsSumsInput = addInput(rhsSums, mapToColumnDim);
       indexOfLhsZpInput = addInput(lhsZp, mapToNone);
     }
     if (!lhsZpIsConstantZero && !rhsZpIsConstantZero) {
       Value lhsZpTimesRhsZp = builder.create<arith::MulIOp>(lhsZp, rhsZp);
+
+      unsigned colIndex = batch ? 2 : 1;
       Value kSize = rewriter.create<arith::IndexCastOp>(
-          loc, accElTy, builder.create<tensor::DimOp>(lhs, 1));
+          loc, accElTy, builder.create<tensor::DimOp>(lhs, colIndex));
```

Comment:
Can we embed `colIndex` into the builder? My point is:

1. It is only used in the builder
2. The `batch ? 2 : 1` have enough readability.

Reply:
Makes sense.